test/torch_optimizer_test.py

```diff
@@ -317,6 +356,80 @@ def test_standard_optimizer(device, communication_type, kwargs):
         test_mse[-3:].max() < error_threshold*problem_builder.noise_level**2
     ), "Train MSE in the last three epochs doesn't coverge."
 
+hierarchical_model_scenarios = []
+hierarchical_model_scenarios.append(
+    pytest.param("CPU", bf.CommunicationType.neighbor_allreduce, {"ATC": False},
+                 id="AWC Neighbor Allreduce on CPU",
+                 marks=pytest.mark.skip(reason="AWC doesn't converge for hierarchical model.")))
+hierarchical_model_scenarios.append(
+    pytest.param("CPU", bf.CommunicationType.neighbor_allreduce, {"ATC": True},
+                 id="ATC Neighbor Allreduce on CPU",
+                 marks=pytest.mark.skip(reason="ATC doesn't converge for hierarchical model.")))
+hierarchical_model_scenarios.append(
+    pytest.param("CPU", "gradient.allreduce", {}, id="Gradient Allreduce on CPU"))
+hierarchical_model_scenarios.append(
+    pytest.param("CPU", "win.put", {}, id="Window put on CPU",
+                 marks=pytest.mark.skip(reason="Multiple win_put optimizer tests will fail")))
+if TEST_ON_GPU:
+    hierarchical_model_scenarios.append(
+        pytest.param("GPU", bf.CommunicationType.neighbor_allreduce, {"ATC": False},
+                     id="AWC Neighbor Allreduce on GPU",
+                     marks=pytest.mark.skip(reason="AWC doesn't converge for hierarchical model.")))
+    hierarchical_model_scenarios.append(
+        pytest.param("GPU", bf.CommunicationType.neighbor_allreduce, {"ATC": True},
+                     id="ATC Neighbor Allreduce on GPU",
+                     marks=pytest.mark.skip(reason="ATC doesn't converge for hierarchical model.")))
+    hierarchical_model_scenarios.append(
+        pytest.param("GPU", "gradient.allreduce", {}, id="Gradient Allreduce on GPU"))
+    hierarchical_model_scenarios.append(
+        pytest.param("GPU", "win.put", {}, id="Window put on GPU",
+                     marks=pytest.mark.skip(reason="Multiple win_put optimizer tests will fail")))
+@pytest.mark.parametrize("device,communication_type,kwargs", hierarchical_model_scenarios)
+def test_optimizer_for_hierarchical_model(device, communication_type, kwargs):
+    atc_style = kwargs["ATC"] if "ATC" in kwargs else False
+    error_threshold = kwargs["error_threshold"] if "error_threshold" in kwargs else 1.5
+
+    problem_builder, train_dataloader, test_dataloader, model, optimizer, num_epochs = \
+        problem_setup(HierarchicalLinearNet)
+
+    isCUDA = device=="GPU"
```

Comment:
This repeats many times and usage should always be the same. So better to create a global util function:

def pin_model_to_device(device):
    ....

Reply:
Done.